probe/endpoint/ebpf.go

```diff
@@ -0,0 +1,285 @@
+package endpoint
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"net"
+	"os"
+	"strconv"
+	"strings"
+	"sync"
+	"syscall"
+	"unsafe"
+
+	log "github.com/Sirupsen/logrus"
+	bpflib "github.com/kinvolk/gobpf-elf-loader/bpf"
+)
+
+import "C"
+
+var byteOrder binary.ByteOrder
+
+func init() {
+	var i int32 = 0x01020304
+	u := unsafe.Pointer(&i)
+	pb := (*byte)(u)
+	b := *pb
+	if b == 0x04 {
+		byteOrder = binary.LittleEndian
+	} else {
+		byteOrder = binary.BigEndian
+	}
+}
+
+type EventType uint32
+
+// These constants should be in sync with the equivalent definitions in the ebpf program.
+const (
+	_ EventType = iota
+	EventConnect
+	EventAccept
+	EventClose
+)
+
+func (e EventType) String() string {
+	switch e {
+	case EventConnect:
+		return "connect"
+	case EventAccept:
+		return "accept"
+	case EventClose:
+		return "close"
+	default:
+		return "unknown"
+	}
+}
+
+// tcpEvent should be in sync with the struct in the ebpf maps.
+type tcpEvent struct {
+	// Timestamp must be the first field, the sorting depends on it
+	Timestamp uint64
+
+	Cpu   uint64
```

Comment:
This should be `CPU` apparently (see https://circleci.com/gh/alepuccetti/scope/51?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link)